homeassistant/components/group/__init__.py

```diff
@@ -222,10 +223,12 @@ def groups_with_entity(hass: HomeAssistant, entity_id: str) -> list[str]:
     return groups
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up all groups found defined in the configuration."""
     if (component := hass.data.get(DOMAIN)) is None:
         component = hass.data[DOMAIN] = EntityComponent(_LOGGER, DOMAIN, hass)
+    if TYPE_CHECKING:  # Can't be None - see previous line
+        assert component is not None
```

Comment:
This is a mypy bug I think...

Reply:
A suggested workaround, without needing all kinds of assertions:

```py
    if DOMAIN not in hass.data:
        hass.data[DOMAIN] = EntityComponent(_LOGGER, DOMAIN, hass)

    component = hass.data[DOMAIN]
```

Reply:
You're right - that's a much nicer workaround. Done